packages/okta-react/README.md

```diff
@@ -314,7 +314,7 @@ These options are used by `Security` to configure the [Auth service][]. The most
 - **onAuthRequired** *(optional)* - callback function. Called when authentication is required. If this is not supplied, `okta-react` redirects to Okta. This callback will receive `authService` as the first function parameter. This is triggered when:
     1. [login](#authserviceloginfromuri-additionalparams) is called
     2. A `SecureRoute` is accessed without authentication
-- **onSessionExpired** *(optional)* - callback function. Called when the Okta SSO session has expired or was ended outside of the application. This SDK adds a default handler which will call [login](#authserviceloginfromuri-additionalparams) to initiate a login flow. Passing a function here will disable the default handler.
+- **onSessionExpired** *(optional)* - callback function. Called when the Okta SSO session has expired or was ended outside of the application. This SDK adds a default handler which will call [clearAuthState](##authserviceclearauthstate) to clear authState in context. Passing a function here will disable the default handler.
```

Comment:
I thought we had decided that the default handler would do nothing, and we'd separately listen to the expired event to call clearAuthState

Reply:
I think this change should make sense: https://github.com/okta/okta-oidc-js/pull/848/commits/7311f47c92cf933410f0e7db13164b46d1ef9283

Reply:
I believe the effect here is that if the accessToken cannot be renewed, the idToken will also be removed. This *might* be the right logical choice, but there could be a timing error if the idToken is checked first. If we want to invalidate all tokens when accessToken renew fails, we would need to make sure to always try to retrieve accessToken before idToken. This is kind of subtle and error prone. Maybe a better choice is to make default `isAuthenticated` logic only true if BOTH idToken and accessToken are present.

Reply:
That change would definitely be a major version change - discussing that change is what led to the idea of removing isAuthenticated entirely

I also don't think that renewal failure of the accessToken in any way invalidates the idToken.  That requires expiration or revocation.

Reply:
https://github.com/okta/okta-auth-js/blob/master/packages/okta-auth-js/lib/browser/browser.js#L184
Looks like based on current `auth-js` implementation, `onSessionExired` callback will only be triggered on accessToken renew failure, then it would make sense to just clear `accessToken` in context.

But how should we handle `idToken` renew failure and `isAuthenticated` state. I think the purpose for this fix is to update the whole `authState` to keep user out of `SecureRoute`.

Reply:
No - the purpose is to have authState be accurate.  If only accessToken is invalid, then authState is updated.  (updateAuthState instead of clearAuthState).

Reply:
it might will cause infinite loop. error -> updateAuthState -> error -> ...
might need some thoughts on when onSessionExpired be triggered first, imo.